dnf/transaction_sr.py

```diff
@@ -341,7 +344,7 @@ def _replay_pkg_action(self, pkg_data):
         elif action in ("Upgraded", "Downgraded", "Reinstalled", "Removed", "Obsoleted"):
             query = query.installed()
 
-            if not query:
+            if not query and not self._history_mode:
```

Comment:
For this case, if `history_mode` is on, the only difference is the warning, as the `query` is empty, so even if we don't `return` in this `if` branch, nothing else will happen anyway.